src/test/rules/terComputedPropertySpacingRuleTests.ts

```diff
@@ -0,0 +1,167 @@
+import { RuleTester, Failure, Position, dedent } from './ruleTester';
+
+const ruleTester = new RuleTester('ter-computed-property-spacing');
+
+// Change this function to better test the rule. In some cases the message never changes so we
+// can avoid passing it in. See other rule tests for examples.
+function expecting(errors: ['yes-before' | 'yes-after' | 'no-before' | 'no-after', string, number, number][]): Failure[] {
+  return errors.map((err) => {
+    const token = err[1];
+    let message;
+
+    if (err[0] === 'yes-before') {
+      message = `A space is required before '${token}'.`;
+    } else if (err[0] === 'yes-after') {
+      message = `A space is required after '${token}'.`;
+    } else if (err[0] === 'no-before') {
+      message = `There should be no space before '${token}'.`;
+    } else {
+      message = `There should be no space after '${token}'.`;
+    }
+
+    return {
+      failure: message,
+      startPosition: new Position(1, err[2]),
+      endPosition: new Position(1, err[3])
+    };
+  });
+}
+
+ruleTester.addTestGroup('never-valid', 'when never, no spaces passes', [
+  {
+    code: dedent`
+      obj[foo]
+    `
+  },
+  {
+    code: dedent`
+      obj['foo']
+    `
+  },
+  {
+    code: dedent`
+      var x = {[b]: a}
+    `
+  },
+  {
+    code: dedent`
+      obj[foo[bar]]
+    `
+  }
+]);
+
+ruleTester.addTestGroup('never-incorrect', 'when never, spaces fail', [
```

Comment:
Are these all the tests they have in this file: https://github.com/eslint/eslint/blob/master/tests/lib/rules/computed-property-spacing.js? Seems kinda short.